services/registration/src/routes/branch.ts

```diff
@@ -1,28 +1,42 @@
 import { asyncHandler } from "@api/common";
 import express from "express";
 
+import { BranchModel } from "../models/branch";
+
 export const branchRouter = express.Router();
 
 branchRouter.route("/").get(
   asyncHandler(async (req, res) => {
-    res.send();
+    const branches = await BranchModel.find({});
+    return res.send(branches);
   })
 );
 
 branchRouter.route("/").post(
   asyncHandler(async (req, res) => {
-    res.send();
+    const new_branch = await BranchModel.create({
+      name: req.body.name,
+      type: req.body.type,
+      settings: req.body.settings,
+    });
+    return res.send(new_branch);
   })
 );
 
 branchRouter.route("/:id").get(
   asyncHandler(async (req, res) => {
-    res.send();
+    const new_branch = await BranchModel.find({ _id: req.query.id });
+    return res.send(new_branch);
   })
 );
 
-branchRouter.route("/:id").put(
+branchRouter.route("/:id").patch(
   asyncHandler(async (req, res) => {
-    res.send();
+    const updated_branch = await BranchModel.findByIdAndUpdate(req.params.id, {
```

Comment:
Here, to get the new document, you have to set `{ new: true }`. See [here](https://stackoverflow.com/questions/30419575/mongoose-findbyidandupdate-not-returning-correct-model)